### ISSUE TITLE
Possible improvements on not supported & invalid regexes

### ISSUE THREAD

**Issue by Stranger6667** [Zac-HD/hypothesis-jsonschema#75]
In web APIs, users often use regular expressions syntax supported by their backend, and sometimes it is not compatible (in some areas) with the one supported by JSON Schema.

For example, this [AWS API](https://github.com/APIs-guru/openapi-directory/blob/master/APIs/amazonaws.com/sso-admin/2020-07-20/openapi.yaml#L2090) uses character classes that are supported by Java, for example, `\p{Alpha}`. It is not supported in Python stdlib `re` module, and currently `hypothesis-jsonschema` [uses](https://github.com/Zac-HD/hypothesis-jsonschema/blob/master/src/hypothesis_jsonschema/_from_schema.py#L413) `st.nothing()` for such cases. In the simplest case, it leads to `Unsatisfiable` as there are no values in this strategy.

But consider an array:

```python
{
    "items": {
        "pattern": "\p{Alpha}",
        "type": "string",
    },
    "maxItems": 50,
    "minItems": 0,
    "type": "array",
}
```

Even though we don't support generating strings for such regular expressions in the schema above, we still can generate an empty array that will match the schema. The same could be applied to optional properties, etc.

The current error output for the schema above:

```
hypothesis.errors.InvalidArgument: Cannot create a collection of max_size=50, because no elements can be drawn from the element strategy nothing()
```

From the user perspective, it will be nice to expose some information about why it happens (the unsupported regex syntax). For cases when we still can generate data without those items/properties, it might be a warning, and for cases when we can't, a better error message will be great (e.g., if there is `minItems: 1`)

What do you think?

P.S. I am pretty sure that I saw a different `InvalidArgument` error that was also connected to drawing from `nothing()` - I will post an update once I find it

**Comment by Zac-HD** [comment]
I think this is two separate issues:

1. Emit a warning when we encounter a regex pattern which is not valid in Python, and 
2. Work around Hypothesis' usual check that list elements is non-empty if max_size is set

**Comment by Zac-HD** [comment]
Fixed by 175213fb6fe40eee8ee9507798de88d4dfdb3b11 and 851d4c75afb1fb0a77654930c7b21c4d27d1beb4 respectively, and released as 0.19 😄

**Comment by Stranger6667** [comment]
Amazing! Thank you! :)
